enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,221 @@
+ a connected user to discover, install, and upgrade tools that are compatible with the current cluster version easily and from a single location.
+
+Each component is responsible for building and publishing its artifacts and registering information regarding supplied binaries.
+Currently, that location is [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/),
+and this makes it difficult for disconnected installations to mirror them.  
+
+`oc` will retrieve binaries from images that package a CLI's artifacts. Currently, we provide disconnected environments 
+cli-artifacts and oc download links that do not require anything outside the cluster.  The goal is to provide the same for other
+binaries/tools/CLIs. Also, these artifacts will be accessible via `oc` as well as through the console via a fileserver.
+
+`oc` will provide the mechanism to provide, list, install, and upgrade OpenShift supported CLIs compatible with   
+a connected cluster. Through a ClusterCLI [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/),
+each CLI will provide necessary information about its provided artifacts, including its description and location within the image.  The ClusterCLI
+for oc, helm, and odo custom resources as well as a CRD controller to manage these will serve as a reference consumer implementation for other CLIs to 
+follow. Currently. the `cli-artifacts` image stream in the `openshift namespace` is shipped with the release payload. This is where the artifacts
+for `oc` will be extracted from. If other CLIs choose to offer artifacts in this way, they can opt in by supplying an artifacts image to be
+shipped with the release payload.
+
+_For this proposal "ClusterCLI" is assumed the name of a CR that will store metadata and information about each CLI_
+
+## Motivation
+
+As more services are created on top of OpenShift, more CLIs are introduced to simplify interaction with these services.
+Some current examples are oc, kubectl, odo, istio, tekton, and knative.  It is difficult for users to discover what tools exist, 
+where to download them from and which version they should download.  We need to simplify as much as possible the interaction
+of services on OpenShift.  We need a mechanism for providing and consuming tools that is simple to add on to as new CLIs are 
+developed from a variety of sources - and this should be specific for each cluster and available with disconnected installs.
+
+## Requirements:
+
+1. No new form of binary distribution or binary creation will be proposed, because we have an existing structure at Red Hat.
+RPMs or images are the only options, and images must be deployed by the RH pipeline via operators. This proposal is for delivering
+CLIs via images, because this will enable offering CLIs offline through mirroring.
```

Comment:
Is the requirement that clients be available using mirroring, or is that an implementation choice?

Reply:
The proposal says

> This proposal is for delivering CLIs via images, because this will enable offering CLIs offline through mirroring.

That combines 2 things: the delivery format and mirroring. Do we have a requirement that CLIs be delivered by images, and as a result of that we get mirroring? Or do we have a requirement that CLI packages be mirrorable, and therefore we're choosing to use images?